src/middlewared/middlewared/plugins/usage.py

```diff
@@ -13,27 +13,35 @@
 
 
 class UsageService(Service):
+
+    FAILED_RETRIES = 3
+
     class Config:
         private = True
 
     async def start(self):
         if (
             await self.middleware.call('system.general.config')
         )['usage_collection']:
-            try:
-                gather = await self.middleware.call('usage.gather')
-                async with aiohttp.ClientSession(
-                    raise_for_status=True
-                ) as session:
-                    await session.post(
-                        'https://usage.freenas.org/submit',
-                        data=gather,
-                        headers={"Content-type": "application/json"},
-                        proxy=os.environ.get("http_proxy"),
-                    )
-            except Exception as e:
-                # We still want to schedule the next call
-                self.logger.error(e)
+            retries = self.FAILED_RETRIES
+            while retries:
+                try:
+                    async with aiohttp.ClientSession(raise_for_status=True) as session:
+                        await session.post(
+                            'https://usage.freenas.org/submit',
+                            data=await self.middleware.call('usage.gather'),
+                            headers={'Content-type': 'application/json'},
+                            proxy=os.environ.get('http_proxy'),
+                        )
+                except Exception as e:
+                    # We still want to schedule the next call
+                    self.middleware.logger.error(e)
+                    retries -= 1
+                    if retries:
+                        self.middleware.logger.debug('Retrying gathering stats after 5 minutes')
+                        await asyncio.sleep(300)
```

Comment:
5 minutes is too low, we talked about an hour. Make it at least 30 minutes please.

Reply:
No problem, we can have it at 30 minutes. It was not mentioned in the ticket at all so defaulted to 5 minutes but yes 30-60 min is indeed more reasonable.